File: android/app/src/main/java/com/freelancekar/app/ui/screens/ResultsScreen.kt
package com.freelancekar.app.ui.screens

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.horizontalScroll
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.unit.dp
import com.freelancekar.app.data.*
import com.freelancekar.app.ui.AppViewModel
import com.freelancekar.app.ui.components.*
import com.freelancekar.app.ui.theme.FkColor
import com.freelancekar.app.ui.theme.FkType

private val SORTS: List<Pair<String, Sort>> = listOf(
    "Best match" to Sort.MATCH,
    "Nearest" to Sort.NEAR,
    "Under ₹15k" to Sort.PRICE,
    "4.8★+" to Sort.RATING,
)

@Composable
fun ResultsScreen(state: AppState, vm: AppViewModel) {
    val city = cityOf(state)
    val results = if (state.searching) emptyList() else selectResults(state)
    val active = filterCount(state)

    val countLabel = if (state.searching) "Searching…" else if (results.size == 1) "1 person" else "${results.size} people"
    val emptyNote = if (active > 0) {
        "Your filters are narrow for ${placeLabel(state)}. Widen the budget or distance and try again."
    } else {
        "Nobody in ${placeLabel(state)} matches \"${state.searchLabel.ifEmpty { "this" }}\" right now. Post the job and freelancers come to you."
    }

    Column(Modifier.fillMaxSize().verticalScroll(rememberScrollState())) {
        Row(
            Modifier.fillMaxWidth().padding(horizontal = 20.dp).padding(top = 16.dp),
            verticalAlignment = Alignment.CenterVertically,
            horizontalArrangement = Arrangement.spacedBy(11.dp),
        ) {
            BackButton { vm.go(Screen.EXPLORE) }
            Row(
                Modifier
                    .weight(1f)
                    .clip(RoundedCornerShape(12.dp))
                    .background(FkColor.card)
                    .border(1.dp, FkColor.line09, RoundedCornerShape(12.dp))
                    .padding(horizontal = 13.dp, vertical = 11.dp),
                verticalAlignment = Alignment.CenterVertically,
            ) {
                Text("⌕", style = FkType.meta.copy(color = FkColor.text45))
                Spacer(Modifier.width(9.dp))
                Text(
                    state.searchLabel.ifEmpty { "Reel editor" },
                    style = FkType.bodySm.copy(color = FkColor.text.copy(alpha = 0.8f)),
                    maxLines = 1,
                )
            }
            Box(
                Modifier.clickable { vm.openFilters() },
            ) {
                Text(
                    "⚙",
                    style = FkType.h3.copy(color = if (active > 0) FkColor.lime else FkColor.text60),
                )
                if (active > 0) {
                    Box(
                        Modifier
                            .align(Alignment.TopEnd)
                            .offset(x = 6.dp, y = (-4).dp)
                            .clip(RoundedCornerShape(50))
                            .background(FkColor.lime)
                            .padding(horizontal = 4.dp, vertical = 1.dp),
                    ) {
                        Text(active.toString(), style = FkType.monoSm.copy(color = FkColor.onLime, fontSize = androidx.compose.ui.unit.TextUnit(8.5f, androidx.compose.ui.unit.TextUnitType.Sp)))
                    }
                }
            }
        }

        Row(
            Modifier.horizontalScroll(rememberScrollState()).padding(horizontal = 20.dp).padding(top = 14.dp),
            horizontalArrangement = Arrangement.spacedBy(8.dp),
        ) {
            SORTS.forEach { (name, key) ->
                Chip(on = state.sort == key, onClick = { vm.setSort(key) }, text = name)
            }
        }

        Row(
            Modifier.fillMaxWidth().padding(horizontal = 20.dp).padding(top = 16.dp),
            horizontalArrangement = Arrangement.SpaceBetween,
            verticalAlignment = Alignment.Bottom,
        ) {
            Text("$countLabel in $city", style = FkType.h4)
            Text(if (state.sort == Sort.MATCH) "MATCH SCORED" else "FILTERED", style = FkType.mono.copy(color = FkColor.text40))
        }

        if (state.searching) {
            Column(Modifier.padding(horizontal = 20.dp).padding(top = 14.dp, bottom = 26.dp), verticalArrangement = Arrangement.spacedBy(11.dp)) {
                repeat(3) {
                    Row(
                        Modifier
                            .fillMaxWidth()
                            .clip(RoundedCornerShape(18.dp))
                            .background(FkColor.card)
                            .border(1.dp, FkColor.line07, RoundedCornerShape(18.dp))
                            .padding(14.dp),
                        horizontalArrangement = Arrangement.spacedBy(12.dp),
                    ) {
                        Box(
                            Modifier
                                .size(48.dp)
                                .clip(RoundedCornerShape(14.dp))
                                .background(FkColor.skeletonLg),
                        )
                        Column(Modifier.weight(1f).padding(top = 3.dp), verticalArrangement = Arrangement.spacedBy(9.dp)) {
                            Box(Modifier.fillMaxWidth(0.58f).height(11.dp).clip(RoundedCornerShape(4.dp)).background(FkColor.skeletonLg))
                            Box(Modifier.fillMaxWidth(0.82f).height(9.dp).clip(RoundedCornerShape(4.dp)).background(FkColor.skeletonSm))
                            Box(Modifier.fillMaxWidth(0.44f).height(9.dp).clip(RoundedCornerShape(4.dp)).background(FkColor.skeletonSm))
                        }
                    }
                }
            }
        } else if (results.isEmpty()) {
            Column(
                Modifier.fillMaxWidth().padding(horizontal = 30.dp).padding(vertical = 30.dp),
                horizontalAlignment = Alignment.CenterHorizontally,
            ) {
                Text("⌕", style = FkType.h2.copy(color = FkColor.text28))
                Spacer(Modifier.height(14.dp))
                Text("Nobody matches this yet", style = FkType.screenTitle)
                Spacer(Modifier.height(8.dp))
                Text(emptyNote, style = FkType.bodySm.copy(color = FkColor.text50), textAlign = androidx.compose.ui.text.style.TextAlign.Center)
                Spacer(Modifier.height(20.dp))
                Column(verticalArrangement = Arrangement.spacedBy(9.dp)) {
                    PrimaryButton({ vm.clearAll() }, text = "Clear filters")
                    OutlineButton({ vm.go(Screen.EXPLORE) }, text = "Post the job instead")
                }
            }
        } else {
            Column(Modifier.padding(horizontal = 20.dp).padding(top = 14.dp, bottom = 26.dp), verticalArrangement = Arrangement.spacedBy(11.dp)) {
                results.forEachIndexed { i, p ->
                    Column(
                        Modifier
                            .fillMaxWidth()
                            .clip(RoundedCornerShape(18.dp))
                            .background(FkColor.card)
                            .border(1.dp, if (i == 0) FkColor.lime else FkColor.line08, RoundedCornerShape(18.dp))
                            .padding(14.dp),
                    ) {
                        Row(
                            Modifier.fillMaxWidth().clickable { vm.openPro(p.id) },
                            horizontalArrangement = Arrangement.spacedBy(12.dp),
                        ) {
                            Stripe(Modifier.size(48.dp).clip(RoundedCornerShape(14.dp)).border(1.dp, FkColor.line09, RoundedCornerShape(14.dp)))
                            Column(Modifier.weight(1f), verticalArrangement = Arrangement.spacedBy(4.dp)) {
                                Row(verticalAlignment = Alignment.CenterVertically) {
                                    Text(p.name, style = FkType.h4.copy(fontSize = androidx.compose.ui.unit.TextUnit(14.5f, androidx.compose.ui.unit.TextUnitType.Sp)))
                                    if (p.verified) {
                                        Spacer(Modifier.width(6.dp))
                                        Text("✔", style = FkType.metaXs.copy(color = FkColor.lime))
                                    }
                                    Spacer(Modifier.weight(1f))
                                    Text("${p.match}%", style = FkType.mono.copy(color = if (i == 0) FkColor.lime else FkColor.text55))
                                }
                                Text(p.services, style = FkType.meta.copy(color = FkColor.text55))
                                Row(horizontalArrangement = Arrangement.spacedBy(9.dp)) {
                                    Text("★ ${p.rating}", style = FkType.meta.copy(color = FkColor.amber))
                                    Text("${p.jobs} jobs", style = FkType.meta.copy(color = FkColor.text40))
                                    Text(p.distance, style = FkType.meta.copy(color = FkColor.text40))
                                }
                            }
                        }
                        Row(
                            Modifier.fillMaxWidth().padding(top = 11.dp).border(0.dp, FkColor.line07).padding(top = 0.dp),
                            horizontalArrangement = Arrangement.SpaceBetween,
                            verticalAlignment = Alignment.CenterVertically,
                        ) {
                            Column {
                                Text(p.price, style = FkType.cardTitle)
                                Text(p.note, style = FkType.metaXs.copy(color = FkColor.text45))
                            }
                            Row(horizontalArrangement = Arrangement.spacedBy(8.dp)) {
                                InlineOutlineButton({ vm.messageFromResults(p) }, text = "Message")
                                InlineButton({ vm.openHire(p.id) }, text = "Hire")
                            }
                        }
                    }
                }
            }
        }
    }
}

File: ios/FreelanceKar/Views/Screens/ResultsScreen.swift
import SwiftUI

struct ResultsScreen: View {
    @EnvironmentObject var store: AppStore
    private let sorts: [(String, Sort)] = [("Best match", .match), ("Nearest", .near), ("Under ₹15k", .price), ("4.8★+", .rating)]

    var body: some View {
        let s = store.state
        let city = Selectors.cityOf(s)
        let results = s.searching ? [] : Selectors.selectResults(s)
        let active = Selectors.filterCount(s)
        let countLabel = s.searching ? "Searching…" : (results.count == 1 ? "1 person" : "\(results.count) people")
        let emptyNote = active > 0
            ? "Your filters are narrow for \(Selectors.placeLabel(s)). Widen the budget or distance and try again."
            : "Nobody in \(Selectors.placeLabel(s)) matches “\(s.searchLabel.isEmpty ? "this" : s.searchLabel)” right now. Post the job and freelancers come to you."

        ScreenShell {
            VStack(alignment: .leading, spacing: 0) {
                HStack(spacing: 11) {
                    BackButton { store.go(.explore) }
                    HStack(spacing: 9) {
                        Image(systemName: "magnifyingglass").font(.system(size: 13)).foregroundStyle(FK.text45)
                        Text(s.searchLabel.isEmpty ? "Reel editor" : s.searchLabel)
                            .font(.system(size: 12.5))
                            .foregroundStyle(Color(hex: 0xF2F2F0, opacity: 0.8))
                            .lineLimit(1)
                        Spacer()
                    }
                    .padding(.horizontal, 13).padding(.vertical, 11)
                    .background(FK.card)
                    .clipShape(RoundedRectangle(cornerRadius: 12, style: .continuous))
                    .overlay(RoundedRectangle(cornerRadius: 12).stroke(FK.line09, lineWidth: 1))

                    Button { store.openFilters() } label: {
                        ZStack(alignment: .topTrailing) {
                            Image(systemName: "slider.horizontal.3")
                                .font(.system(size: 15))
                                .foregroundStyle(active > 0 ? FK.lime : FK.text60)
                            if active > 0 {
                                Text("\(active)")
                                    .font(.system(size: 8.5, weight: .semibold, design: .monospaced))
                                    .foregroundStyle(FK.onLime)
                                    .padding(.horizontal, 3).padding(.vertical, 1)
                                    .background(FK.lime)
                                    .clipShape(Capsule())
                                    .offset(x: 8, y: -6)
                            }
                        }
                    }
                }
                .padding(.horizontal, 20)
                .padding(.top, 16)

                ScrollView(.horizontal, showsIndicators: false) {
                    HStack(spacing: 8) {
                        ForEach(sorts, id: \.1) { name, key in
                            Chip(title: name, on: s.sort == key) { store.setSort(key) }
                        }
                    }.padding(.horizontal, 20)
                }
                .padding(.top, 14)

                HStack(alignment: .lastTextBaseline) {
                    Text("\(countLabel) in \(city)").font(.fkH4)
                    Spacer()
                    Text(s.sort == .match ? "MATCH SCORED" : "FILTERED")
                        .font(.system(size: 11, design: .monospaced))
                        .foregroundStyle(FK.text40)
                }
                .padding(.horizontal, 20)
                .padding(.top, 16)

                if s.searching {
                    VStack(spacing: 11) {
                        ForEach(0..<3, id: \.self) { _ in
                            HStack(spacing: 12) {
                                RoundedRectangle(cornerRadius: 14).fill(FK.skeletonLg).frame(width: 48, height: 48)
                                VStack(alignment: .leading, spacing: 9) {
                                    RoundedRectangle(cornerRadius: 4).fill(FK.skeletonLg).frame(width: 140, height: 11)
                                    RoundedRectangle(cornerRadius: 4).fill(FK.skeletonSm).frame(width: 200, height: 9)
                                    RoundedRectangle(cornerRadius: 4).fill(FK.skeletonSm).frame(width: 100, height: 9)
                                }
                            }
                            .padding(14)
                            .background(FK.card)
                            .clipShape(RoundedRectangle(cornerRadius: 18, style: .continuous))
                            .opacity(0.7)
                        }
                    }
                    .padding(.horizontal, 20).padding(.top, 14)
                } else if results.isEmpty {
                    VStack(spacing: 0) {
                        Text("⌕").font(.system(size: 24)).foregroundStyle(FK.text28)
                        Text("Nobody matches this yet").font(.fkScreenTitle).padding(.top, 14)
                        Text(emptyNote).font(.fkBodySm).foregroundStyle(FK.text50).multilineTextAlignment(.center).padding(.top, 8)
                        VStack(spacing: 9) {
                            PrimaryButton(title: "Clear filters") { store.clearAll() }
                            OutlineButton(title: "Post the job instead") { store.go(.explore) }
                        }.padding(.top, 20)
                    }
                    .padding(.horizontal, 30).padding(.vertical, 30)
                } else {
                    VStack(spacing: 11) {
                        ForEach(Array(results.enumerated()), id: \.element.id) { i, p in
                            ResultCard(pro: p, highlighted: i == 0)
                        }
                    }
                    .padding(.horizontal, 20).padding(.top, 14).padding(.bottom, 26)
                }
            }
        }
    }
}

private struct ResultCard: View {
    @EnvironmentObject var store: AppStore
    var pro: Pro
    var highlighted: Bool

    var body: some View {
        VStack(spacing: 0) {
            Button { store.openPro(pro.id) } label: {
                HStack(spacing: 12) {
                    Stripe(cornerRadius: 14).frame(width: 48, height: 48)
                        .overlay(RoundedRectangle(cornerRadius: 14).stroke(FK.line09, lineWidth: 1))
                    VStack(alignment: .leading, spacing: 4) {
                        HStack(spacing: 6) {
                            Text(pro.name).font(.system(size: 14.5, weight: .semibold))
                            if pro.verified {
                                Text("✔").font(.system(size: 11)).foregroundStyle(FK.lime)
                            }
                            Spacer()
                            Text("\(pro.match)%")
                                .font(.system(size: 11, weight: .semibold, design: .monospaced))
                                .foregroundStyle(highlighted ? FK.lime : FK.text55)
                        }
                        Text(pro.services).font(.fkMeta).foregroundStyle(FK.text55)
                        HStack(spacing: 9) {
                            Text("★ \(pro.rating, specifier: "%.1f")").foregroundStyle(FK.amber)
                            Text("\(pro.jobs) jobs").foregroundStyle(FK.text40)
                            Text(pro.distance).foregroundStyle(FK.text40)
                        }.font(.system(size: 11, weight: .medium))
                    }
                }
            }
            .foregroundStyle(FK.text)

            HStack {
                VStack(alignment: .leading, spacing: 0) {
                    Text(pro.price).font(.fkCardTitle)
                    Text(pro.note).font(.system(size: 10.5)).foregroundStyle(FK.text45)
                }
                Spacer()
                HStack(spacing: 8) {
                    InlineOutlineButton(title: "Message") { store.messageFromResults(pro) }
                    InlineButton(title: "Hire") { store.openHire(pro.id) }
                }
            }
            .padding(.top, 11)
            .overlay(Rectangle().frame(height: 1).foregroundStyle(FK.line07), alignment: .top)
            .padding(.top, 12)
        }
        .padding(14)
        .background(FK.card)
        .clipShape(RoundedRectangle(cornerRadius: 18, style: .continuous))
        .overlay(RoundedRectangle(cornerRadius: 18).stroke(highlighted ? FK.lime : FK.line08, lineWidth: 1))
    }
}
